src/contrib/cluster/Akka.DistributedData/Replicator.cs

```diff
@@ -1422,14 +1435,14 @@ protected override ImmutableArray<Address> AllNodes
                     return allNodes;
                 }
             }
-                
+
 
             protected override DeltaPropagation CreateDeltaPropagation(ImmutableDictionary<string, (IReplicatedData data, long from, long to)> deltas)
             {
                 // Important to include the pruning state in the deltas. For example if the delta is based
                 // on an entry that has been pruned but that has not yet been performed on the target node.
                 var newDeltas = deltas
-                    .Where(x => !Equals(x.Value.data, DeltaPropagation.NoDeltaPlaceholder))
+                    .Where(x => !ReferenceEquals(x.Value.data, DeltaPropagation.NoDeltaPlaceholder))
```

Comment:
Why the `ReferenceEquals` change here?

Reply:
because it makes a lot more sense, `NoDeltaPlaceholder` is a reference to a singleton instance of `NoDelta`, it makes more sense to compare it by reference